### PR TITLE
Gomspace HOOTL Bug

### LINKED ISSUE #491: GomspaceCheckoutCase is broken in HOOTL
There is an error message 

```
Power cycled outputs could not turn on after 600 cycles (1 minute)
Could not update power_cycle_output0
Could not update power_cycle_output1
Could not update power_cycle_output2
Could not update power_cycle_output3
Could not update power_cycle_output4
Could not update power_cycle_output5
```

even when using `fsw_native_leader_realtime`. Need to check why this is, and if it's happening in HITL...

### PR DESCRIPTION
**Pull request by fatimayousuf**
# Gomspace HOOTL Bug

Fixes #491 

### Summary of changes
- Make the power cycle period a state field so that the checkout case can easily read it
- Thee checkout case should check that powercycling happens every period. 

### Testing
I ran the checkout case with fsw_native_leader, fsw_native_leader_realtime, and fsw_teensy36_hitl_leader. The power cycling bug did not show up.

### Constants
I got rid of the gomspace period tracked constant, since I turned it into a state field instead.

### CODE REVIEW

File: src/fsw/FCCode/GomspaceController.hpp [hunk 1]
```diff
@@ -117,7 +117,8 @@ class GomspaceController : public TimedControlTask<void> {
     #endif
     // If PAN::one_day_ccno is very short due to the SPEEDUP flag, ensure the period is positive
     // to prevent a divide-by-zero error.
```
Comment (tanishqaggarwal) [inline src/fsw/FCCode/GomspaceController.hpp:119]: Can you move this comment into GomspaceController.cpp near the computation?

File: ptest/cases/gomspace_checkout_case.py [hunk 2]
```diff
@@ -88,6 +88,8 @@ def run_case_singlesat(self):
                                                                + str(i) + "_cmd", "true"))
```
Comment (tanishqaggarwal) [inline ptest/cases/gomspace_checkout_case.py:88]: Can you revert back lines 162-164 (adding back in `TestCaseFailure`)?

File: src/fsw/FCCode/GomspaceController.cpp [hunk 3]
```diff
@@ -74,6 +74,9 @@ GomspaceController::GomspaceController(StateFieldRegistry &registry, unsigned in
     heater_sr(),
     heater_f("gomspace.heater", heater_sr),
 
+    period_sr(),
+    period_f("gomspace.period", period_sr),
```
Comment (tanishqaggarwal) [inline src/fsw/FCCode/GomspaceController.cpp:78]:
```suggestion
    period_f("gomspace.powercycle_period", period_sr),
```
Just to be more descriptive with the state field name.

File: ptest/cases/gomspace_checkout_case.py [hunk 4]
```diff
@@ -88,6 +88,8 @@ def run_case_singlesat(self):
...
                                   for i in range(1, 7)]
         self.cycle()
+        while int(self.read_state("pan.cycle_no")) % int(self.read_state("gomspace.period")) != 0:
```
Comment (tanishqaggarwal) [inline ptest/cases/gomspace_checkout_case.py:91]:
```suggestion
        while int(self.read_state("pan.cycle_no")) % int(self.read_state("gomspace.powercycle_period")) != 0:
```

**Review by tanishqaggarwal — CHANGES_REQUESTED** [review]
You might have to recompile telemetry (`pio run -e gsw_telem_info_generator ; .pio/build/gsw_telem_info_generator/program telemetry`)

File: ptest/cases/gomspace_checkout_case.py (hunk 2 above)
Comment (fatimayousuf) [reply ptest/cases/gomspace_checkout_case.py:88]: Ah sorry I didn't see that

**Review by tanishqaggarwal — APPROVED** [review]

**Pull request merged**